plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/metastore/DeltaLakeTableMetadataScheduler.java

```diff
@@ -0,0 +1,230 @@
+    private ScheduledExecutorService scheduler;
+
+    @Inject
+    public DeltaLakeTableMetadataScheduler(
+            HiveMetastoreFactory hiveMetastoreFactory,
+            NodeManager nodeManager,
+            TypeManager typeManager,
+            @TableParameterLengthLimit int tableParameterLengthLimit,
+            DeltaLakeConfig config)
+    {
+        this.hiveMetastoreFactory = requireNonNull(hiveMetastoreFactory, "hiveMetastoreFactory is null");
+        this.typeManager = requireNonNull(typeManager, "typeManager is null");
+        this.tableParameterLengthLimit = tableParameterLengthLimit;
+        this.storeTableMetadataThreads = config.getStoreTableMetadataThreads();
+        this.storeTableMetadataInterval = config.getStoreTableMetadataInterval();
+        requireNonNull(nodeManager, "nodeManager is null");
+        this.enabled = config.isStoreTableMetadataEnabled() && nodeManager.getCurrentNode().isCoordinator();
+    }
+
+    public void putAll(Map<SchemaTableName, UpdateInfo> tableParameters)
+    {
+        updateInfos.putAll(tableParameters);
+    }
+
+    @PostConstruct
+    public void start()
+    {
+        if (enabled) {
+            int threads = storeTableMetadataThreads.orElse(0);
+            executor = threads == 0 ? newDirectExecutorService() : newFixedThreadPool(threads, threadsNamed("store-table-metadata-%s"));
+            scheduler = newSingleThreadScheduledExecutor(daemonThreadsNamed("store-table-metadata"));
+            scheduler.scheduleWithFixedDelay(this::process, 0, storeTableMetadataInterval.toMillis(), MILLISECONDS);
+        }
+    }
+
+    @VisibleForTesting
+    public void process()
+    {
+        Map<SchemaTableName, UpdateInfo> updateTables;
+        synchronized (this) {
+            updateTables = updateInfos.entrySet().stream()
+                    .collect(toImmutableMap(Map.Entry::getKey, Map.Entry::getValue, maxBy(comparing(UpdateInfo::version))));
+            updateInfos.clear();
+        }
+
+        log.debug("Processing %s table(s): %s", updateTables.size(), updateTables.keySet());
+        for (Map.Entry<SchemaTableName, UpdateInfo> entry : updateTables.entrySet()) {
+            executor.execute(() -> updateTable(entry.getKey(), entry.getValue()));
+        }
+    }
+
+    private void updateTable(SchemaTableName schemaTableName, UpdateInfo info)
+    {
+        log.debug("Updating table: '%s'", schemaTableName);
+        HiveMetastore metastore = hiveMetastoreFactory.createMetastore(Optional.of(info.identity));
+        try {
+            metastore.getTable(schemaTableName.getSchemaName(), schemaTableName.getTableName())
+                    .ifPresent(table -> {
+                        Table updatedTable = setTableMetadata(table, info.version, info.schemaString, info.tableComment);
+                        metastore.replaceTable(table.getDatabaseName(), table.getTableName(), updatedTable, buildInitialPrivilegeSet(table.getOwner().orElseThrow()));
```

Comment:
We want compare-and-set semantics. With Glue, this should replace using `com.amazonaws.services.glue.model.UpdateTableRequest#withVersionId`.
With Thrift, this could use table locks.

(See iceberg catalogs implementations)